### DISCUSSION TITLE
"shader_type compute"?

### DISCUSSION THREAD

**Discussion by sammonius** [godotengine/godot-proposals#9091] [Engine Core]
It's really inconvenient that compute shaders need to be written outside the editor in glsl. On top of having to learn another language and find another text editor, there's this entire new API for them that involves turning everything into a PackedByteArray for some reason. Adding a "shader_type compute" where a function called "main" is called is a super obvious fix for all of this.

**Comment by Calinou** [comment]
Thanks for the proposal! Consolidating in https://github.com/godotengine/godot-proposals/issues/1027.
